<?xml version='1.0'?>
<launch>
<arg name="racecar_name" default="racecar"/>
<arg name="make_required" default="true"/>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find deepracer_simulation_environment)/config/racecar_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_manager" pkg="controller_manager" type="spawner" respawn="false" required="$(arg make_required)"
        output="screen" args="left_rear_wheel_velocity_controller right_rear_wheel_velocity_controller
                                            left_front_wheel_velocity_controller right_front_wheel_velocity_controller
                                            left_steering_hinge_position_controller right_steering_hinge_position_controller
                                            joint_state_controller"/>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen" required="$(arg make_required)">
    <remap from="/joint_states" to="/$(arg racecar_name)/joint_states" />
  </node>

</launch>
